django_cloud_tasks/tasks/publisher_task.py

```diff
@@ -27,21 +27,19 @@ def asap(cls, message: dict, attributes: dict[str, str] | None = None):
         return cls.push(task_kwargs=task_kwargs)
 
     def run(self, message: dict, attributes: dict[str, str] | None = None, headers: dict[str, str] | None = None):
-        # Cloud PubSub does not support headers, but we simulate them with prefixed attributes
-        all_attributes = self._build_attributes(attributes=attributes, headers=headers)
+        # Cloud PubSub does not support headers, but we simulate them with a key in the data property
+        self._set_message_headers(message=message, headers=headers)
```

Comment:
Do not change in-place, please.
```suggestion
        message = self._set_message_headers(message=message.copy(), headers=headers)
```